### PR TITLE
 Transpose-Flatten-Concat fusion operator. 

### PR DESCRIPTION
**Pull request by qingqing01**
Support patter:

```
concat_op [ flatten_op(transpose(x1)),  flatten_op(transpose(x2)), flatten_op(transpose(x3)), .... ]
```

like:

default

### CODE REVIEW

**Review by Superjomn — DISMISSED** [review]
LGTM

File: paddle/fluid/CMakeLists.txt [hunk 1]
```diff
@@ -7,5 +7,5 @@ add_subdirectory(recordio)
 add_subdirectory(pybind)
 
 # NOTE: please add subdirectory inference at last.
-add_subdirectory(inference)
-add_subdirectory(train)
+#add_subdirectory(inference)
```
Comment (Superjomn) [inline paddle/fluid/CMakeLists.txt]: recover these

**Review by Superjomn — CHANGES_REQUESTED** [review]
fix the CMakeFiles

File: paddle/fluid/CMakeLists.txt (hunk 1 above)
Comment (qingqing01) [reply paddle/fluid/CMakeLists.txt]: Done, thanks!

**Review by Superjomn — APPROVED** [review]
LGTM

**Comment by qingqing01** [comment]
The doc will update in next PR.

**Pull request merged**